lib/breadcrumbs.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+var BreadcrumbBuffer = function BreadcrumbBuffer(maxBreadcrumbs) {
+  this._buffer = [];
+  this._maxBreadcrums = maxBreadcrumbs;
+};
+
+BreadcrumbBuffer.prototype.add = function add(crumb) {
+  this._buffer.push(crumb);
+  if (this._buffer.length > this._globalOptions.maxBreadcrumbs) {
+    this._buffer.shift();
```

Comment:
Just a heads up that `n` should never exceed `100` (which afaik is the hard limit across different client libs)